Quadrat: Add Episode block pattern

MaggieCabrera opened the pull request:
<!-- Thanks for contributing to our free themes! Please provide as much information as possible with your Pull Request by filling out the following - this helps make reviewing much quicker! -->

#### Changes proposed in this Pull Request:

This PR adds the Episode Block Pattern.

Desktop:

<img width="1214" alt="Screenshot 2021-04-23 at 15 51 46" src="https://user-images.githubusercontent.com/3593343/115881930-8d4c7500-a44c-11eb-8328-c11ccd03eb9a.png">

Mobile:

<img width="396" alt="Screenshot 2021-04-23 at 15 56 12" src="https://user-images.githubusercontent.com/3593343/115881953-90dffc00-a44c-11eb-8a07-90cd9f92e8f4.png">


#### Related issue(s):

Partially addresses https://github.com/Automattic/themes/issues/3642

MaggieCabrera committed:
create episode pattern

MaggieCabrera committed:
added group wrapper for padding

jffng commented on quadrat/sass/_utility.scss:
```diff
@@ -1,7 +1,11 @@
 	}
 }
 
+.has-bakground-no-padding.wp-block-columns.has-background {
```
Spelling nit that would need to be updated in the pattern too.

```suggestion
.has-background-no-padding.wp-block-columns.has-background {
```

MaggieCabrera replied:
great catch!

jffng commented on quadrat/inc/block-patterns.php:
```diff
@@ -117,6 +117,34 @@ function quadrat_register_block_patterns() {
+					'content'    => '<!-- wp:columns {"verticalAlignment":"center","align":"wide","backgroundColor":"darker-blue","textColor":"pink","className":"has-bakground-no-padding"} -->
+					<div class="wp-block-columns alignwide are-vertically-aligned-center has-bakground-no-padding has-pink-color has-darker-blue-background-color has-text-color has-background"><!-- wp:column {"verticalAlignment":"center","width":"40%"} -->
+					<div class="wp-block-column is-vertically-aligned-center" style="flex-basis:40%"><!-- wp:image {"id":1497,"sizeSlug":"full","className":"image-no-margin mb-0"} -->
+					<figure class="wp-block-image size-full image-no-margin mb-0"><img src="http://freethemes.local/wp-content/uploads/2021/04/Introspection-scaled.jpg" alt="' . esc_attr__( 'Illustration of an introspective woman.', 'quadrat') . '" class="wp-image-1497"/></figure>
```
Upload this image to the staging site?

MaggieCabrera replied:
uh, that's the copy paste that got rid of the correct url, thanks for noticing!

MaggieCabrera committed:
correct url for image

MaggieCabrera committed:
typo

jffng commented:
Could this just be a media & text block? 

I tried it quickly and it seems to work out of the box:

Desktop | Mobile
--------- | --------
<img width="1174" alt="Screen Shot 2021-04-23 at 11 57 33 AM" src="https://user-images.githubusercontent.com/5375500/115898080-0084c580-a412-11eb-99d6-b8cb9e8cccfb.png"> | <img width="412" alt="Screen Shot 2021-04-23 at 11 57 48 AM" src="https://user-images.githubusercontent.com/5375500/115898093-02e71f80-a412-11eb-89a8-6c637e483f69.png">

MaggieCabrera commented:
> Could this just be a media & text block?
> 



I did try this, but since I had used a 2x image, the proportions were not correct and I couldn't modify that with that block. Should we be using images at 1x resolution?

MaggieCabrera commented on quadrat/inc/block-patterns.php:
```diff
@@ -117,6 +117,34 @@ function quadrat_register_block_patterns() {
+					<div class="wp-block-column is-vertically-aligned-center" style="flex-basis:50%"><!-- wp:group {"style":{"spacing":{"padding":{"top":"20px","right":"20px","bottom":"20px","left":"20px"}}}} -->
+					<div class="wp-block-group" style="padding-top:20px;padding-right:20px;padding-bottom:20px;padding-left:20px">
+					<!-- wp:heading {"level":3,"textColor":"pink"} -->
+					<h3 class="has-pink-color has-text-color">' . esc_html__( 'Episode 1: How Screens Affect Hormones', 'quadrat') . '</h3>
```
This text is incorrect. I blame copypaste again.

scruffian committed:
update to use media and text

scruffian commented:
I updated the copy and changed to use a media and text block.

scruffian commented on quadrat/sass/_utility.scss:
```diff
@@ -1,7 +1,11 @@
-.image-no-margin {
+.image-no-margin,
+.wp-block-image.image-no-margin {
```
do we still need these changes?

MaggieCabrera replied:
Not anymore, removed!

MaggieCabrera committed:
changed the url of the image, added alt content and em tag

MaggieCabrera committed:
remove unneeded class, added wrapper for bottom padding

MaggieCabrera commented:
I added a group wrapper to add bottom padding (it could be done with a spacer too, but not sure what's the better way) so that if we stack multiples of these patterns there's a space between them. Also updated the url of the image to pull from staging and some other minor fixes. Ready for another round of 👀

scruffian commented:
I'm not sure about using a group block - I accidentally added the pattern again, inside the group block, so the nesting got messy.

MaggieCabrera commented:
> I'm not sure about using a group block - I accidentally added the pattern again, inside the group block, so the nesting got messy.

then maybe just add a 30px spacer after it?

jffng committed:
Scope the 0 padding to text content without a background.

jffng committed:
Remove group, add spacer. Fix local image ref.

jffng committed:
Rebase and fix merge conflict.

jffng commented:
I made a few changes based on your conversation: 

- Removed group block, added a spacer after instead
- Fixed a mobile padding bug and a mediaLink that was pointing to a local path (is the mediaLink attribute even necessary?)
- Rebased 

If it looks good to you all I say 🚢

pbking reviewed (approved):
LGTM! :shipit:
